kratos/utilities/geometry_tester.cpp

```diff
@@ -944,8 +961,74 @@ bool GeometryTesterUtility::VerfiyShapeFunctionsValues(
 /***********************************************************************************/
 /***********************************************************************************/
 
+bool GeometryTesterUtility::VerifyShapeFunctionsSecondDerivativesValues(
+    GeometryType& rGeometry,
+    GeometryType::CoordinatesArrayType& rGlobalCoordinates,
+    std::stringstream& rErrorMessage
+    )
+{
+    GeometryType::CoordinatesArrayType local_coordinates;
+    rGeometry.PointLocalCoordinates( local_coordinates, rGlobalCoordinates );
+
+    Vector f_1 = ZeroVector(rGeometry.size());
+    Vector f_2 = ZeroVector(rGeometry.size());
+    Vector f_3 = ZeroVector(rGeometry.size());
+    Vector f_4 = ZeroVector(rGeometry.size());
+
+    DenseVector<Matrix> DDN_DX, H;
+    Vector ei,ej,f;
+
+    double delta = 1e-1;
+    unsigned int dim = rGeometry.Dimension();
+    rGeometry.ShapeFunctionsSecondDerivatives(DDN_DX,local_coordinates);
+
+    if ( H.size() != rGeometry.size() )
+    {
+        // KLUDGE: While there is a bug in
+        // ublas vector resize, I have to put this beside resizing!!
+        DenseVector<Matrix> temp( rGeometry.size() );
+        H.swap( temp );
+    }
+
+
+    for ( unsigned int i = 0; i < rGeometry.size(); ++i ) H[i].resize(dim, dim, false);
+
+    for (unsigned int i = 0; i<dim;i++){
+        for (unsigned int j = 0; j<dim;j++){
+            ei = ZeroVector(dim);
```

Comment:
Your size mismatches probably originate from here. ```dim``` is a loop variable going from 0 to 2. So you're constructing zero vectors of differing sizes in each iteration.

Just construct the unit vectors here during the iteration instead of always resetting them. This is just a test so you don't need to worry about performance that much.

Reply:
dim is not a loop variable but i and j. dim is always the same

Reply:
ah my bad, nevermind then